pkg/controller/csi_handler.go

```diff
@@ -610,11 +613,21 @@ func (h *csiHandler) saveAttachError(ctx context.Context, va *storage.VolumeAtta
 	logger := klog.FromContext(ctx)
 	logger.V(4).Info("Saving attach error")
 	clone := va.DeepCopy()
-	clone.Status.AttachError = &storage.VolumeError{
+
+	volumeError := &storage.VolumeError{
 		Message: err.Error(),
 		Time:    metav1.Now(),
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.MutableCSINodeAllocatableCount) {
```

Comment:
IMO, you will have to first define external-attacher only feature gates. You can't read feature gates from k8s this way.

See - https://github.com/kubernetes-csi/external-attacher/blob/master/cmd/csi-attacher/main.go#L98 and how other sidecars define sidecar local feature gates.

Reply:
Thank you Hemant 💯 , published new rev doing the needful.